tests/cloner_test.go

```diff
@@ -0,0 +1,133 @@
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if sourcePvc != nil {
+			err = f.DeletePVC(sourcePvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Cloner should clone data from PVC in name space to other PVC in same name space", func() {
+		targetPvc, err := f.CreatePVCFromDefinition(utils.NewPVCDefinition(
+			"target-pvc",
+			"1G",
+			map[string]string{controller.AnnCloneRequest: f.Namespace.Name + "/" + sourcePVCName},
+			nil))
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, targetPvc.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		sourcePod, err := f.FindPodByPrefix(common.CLONER_SOURCE_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+		targetPod, err := f.FindPodByPrefix(common.CLONER_TARGET_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(strings.Compare(sourcePod.Spec.NodeName, targetPod.Spec.NodeName) == 0).To(BeTrue())
+
+		err = f.WaitTimeoutForPodStatus(sourcePod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitTimeoutForPodStatus(targetPod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		// Clone is completed, verify the content matches the source.
+		Expect(verifyTargetContent(f, f.Namespace, targetPvc)).To(BeTrue())
+		// Clean up PVC, the AfterEach will also clean it up, through the Namespace delete.
+		if targetPvc != nil {
+			err = f.DeletePVC(targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Cloner should clone data from PVC in name space to other PVC in different name space", func() {
+		targetNs, err := f.CreateNamespace(f.NsPrefix, map[string]string{
+			framework.NsPrefixLabel: f.NsPrefix,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		f.AddNamespaceToDelete(targetNs)
+		// Create targetPvc in new NS.
+		targetPvc, err := utils.CreatePVCFromDefinition(f.K8sClient, targetNs.Name, utils.NewPVCDefinition(
+			"target-pvc",
+			"1G",
+			map[string]string{controller.AnnCloneRequest: f.Namespace.Name + "/" + sourcePVCName},
+			nil))
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitForPersistentVolumeClaimPhase(f.K8sClient, targetNs.Name, v1.ClaimBound, targetPvc.Name)
+
+		sourcePod, err := f.FindPodByPrefix(common.CLONER_SOURCE_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+		targetPod, err := utils.FindPodByPrefix(f.K8sClient, targetNs.Name, common.CLONER_TARGET_PODNAME, common.CDI_LABEL_SELECTOR)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(strings.Compare(sourcePod.Spec.NodeName, targetPod.Spec.NodeName) == 0).To(BeTrue())
+
+		err = f.WaitTimeoutForPodStatus(sourcePod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitTimeoutForPodStatus(f.K8sClient, targetPod.Name, targetNs.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		// Clone is completed, verify the content matches the source.
```

Comment:
Can we add a check here to verify that all the correct annotations on the PVC exist after the clone is successful.

Reply:
here and L74

Reply:
Done